---
title: "Analysis Report 2: Your Title Here"
author: "Don Francisco"
date: "November 21, 2018"
output: github_document
bibliography: references.bib
csl: bioinformatics.csl
---

# Introduction

Add about 1 page here. Must cite at least 5 peer reviewed articles.

# Methods

## Sample origin and sequencing

### RNA Sequencing Data and Phenotype Data

Li et al. downloaded the transcriptome sequencing data from 68 lung adenocarcinoma patients with validated smoking status from Gene Expression Omnibus (GEO) with the accession number GSE40419 [@li2015rna]. In the 68 patients, both the tumor and paired normal tissue had been sequenced. With 34 patients who had never been smokers and 34 patients who either have smoked or are current smokers, they restricted analysis of the people who either have smoked or are current smokers to patients under 75 in order to ensure comparability in age ranges between both groups. Li et al. downloaded independent RNA-seq data from six patients who have never smoked from GEO with the accession number GSE37765 in order to validate the genes in the nonsmoker group. The age range for these six patients varied from 44 to 70. 

### Differential Gene Expression Analysis 

Li et al. conducted geneotype calling of single nucleotide variants (SNVs) on 136 RNA-seq bam files from 68 individuals [@li2015rna]. This was done in order to confirm the paired relationship of RNA-seq data from both normal and tumor tissue. In a 2012 study, Seo et al. used a clustering method in order to recognize differentially expressed genes in both smoker and nonsmoker patients that had lung adenocarcinoma. They discovered genes with varied expression between normal and tumor tissue across their data samples [@seo2012transcriptional]. Li et al. on the other hand, analyzed differential gene expression using pair-wise comparison between both normal and tumor tissue and nonsmokers and smokers. They calculated pair-wise concordance rates for 9180 pairs total, which includes between individuals and between individual pairs. They found that the there is a paired nature of RNA-seq data from tumor and normal tissue. Using Tophat, Li et al. aligned pair-end RNA-seq reads to human genome assembly Ensembl GRCh37. They used R Bioconductor edgeR to perform the differential expression analysis and applied a general lienar model (lung tissue expression~smoking+smoking:patient+smoking:tissue). This model allowed them to identify genes that were differentially expressed in normal versus tumor tissue in nonsmoker or smoker patients, as well as genes that act differently between smoker and nonsmoker patients. They only kept tags that had at least 1 count per million in at least half of the sample size in the analysis. They noted significant genes as genes with Benjamini-Hochberg adjusted FDF < 0.05 and absolute values of logFC greater than 1. 

### Gene Function Annotation and Pathway Analysis

In order to look into the functional roles and relationships of the genes with different expression in the analysis, Li et al. analyzed the pathway. To analyze the pathway, they submitted the logFC value of significant genes to Ingenuity Pathway Analysis (IPA). 

## Computational

These are the methods that were used to process the sequencing data. Should probably be at least a half of a page. At a very minimum should include citations for biomartr, trimmomatic, and sailfish. Note that these three methods references don't count towards the five references you need to cite in the introduction.

# Results

In addition to a minimum of 4-5 figures/tables (and associated captions), you should include sufficient text in this section to describe what your findings were. Remember that in the results section you just describe what you found, but you don't interpret it - that happens in the discussion.

```{r load-libraries, message = FALSE, echo = FALSE}
# Be sure to install these packages before running this script
# They can be installed either with the intall.packages() function
# or with the 'Packages' pane in RStudio

# load general-use packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")
library("magrittr")

# this package allows for the easy inclusion of literature citations in our Rmd
# more info here: https://github.com/crsh/citr
# and here:
# http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
library("citr")
```

```{r load-data, message = FALSE, echo = FALSE}
# load the dataset from a compressed binary file
# it gets loaded as an object called "joined_table"
# this has 6.2 million rows...so you will need to be thoughtful about
# how you analyze the data so that you don't overwhelm your laptop
load("output/final_compiled_counts/joined_count_data.RData")

# test that it loaded correctly before proceeding
stopifnot(exists("joined_table"))
```

```{r make-summary-table, echo = FALSE}
# There are many many genes in this dataset, so we can
# subset it down to just a few here to look for interesting patterns
# in the most highly expressed
top_15 <- joined_table %>%
  group_by(gender, genename) %>%
  summarize(mean_count = mean(counts_lengthscaledtpm)) %>%
  arrange(desc(mean_count)) %>%
  head(n = 15)

# then we can use the `kable()` function to make a nicely formatted
# markdown table
top_15 %>%
  kable()
```

**Table 1**: The most highly expressed genes in both genders included *SFTPB* and *EEF1A1*.

```{r make-barplot-of-highly-expressed-genes, echo = FALSE}
# this code uses the same data as above, but use it to make a
# barplot - remember geom_col() is just like
# when you use geom_bar(stat = "identity")
top_15 %>%
  ggplot(aes(x = genename,
             y = mean_count,
             fill = gender)) +
    geom_col(position = "dodge")
```

**Figure 1**: Here we show an example figure caption.

```{r make-boxplot-of-highly-expressed-genes, echo = FALSE}
# here we just want to pull out the unique gene names and turn
# them into a vector so we can use it below to make a boxplot
# we use the pull() funtion to get this as a vector, just like
# we did when making histograms several weeks ago
top_genes <- top_15 %>%
  ungroup() %>%
  select(genename) %>%
  unique() %>%
  pull()

# now we need to filter from the full data set again, because
# we don't just want summary data, we want all the data in
# order to make boxplots
joined_table %>%
  filter(genename %in% top_genes) %>%
  ggplot(aes(x = genename,
             y = counts_lengthscaledtpm,
             fill = gender)) +
    geom_boxplot() +
    facet_wrap(~cancer_stage) +
    xlab("Gene Name") +
    ylab("Scaled read counts per gene") +
    ggtitle("Read counts per gene by gender and cancer stage") +
    theme_bw() + # simplifies theme
    theme(axis.text.x = # rotates x axis labels vertically
            element_text(angle = 90,
                         hjust = 1))
```

**Figure 2**: Here we show another example figure caption.
```{r bar-graph-of-three-genes}
# This code creates a graph to look at the 
# comparison of expression values in these 
# three genes. These three genes are looked at 
# because they are the three genes that showed 
# variance between smokers and nonsmokers in the
# Li et al. study. The highest expression value 
# will be looked at in more detail in terms of its
# expression values in cancer tissue vs. normal tissue
# specific stages, and nonsmokers vs smokers. 
joined_table %>%
  filter(genename == "SERPIND1" | 
           genename == "TNNT1" | 
           genename == "PCSK2") %>%
  summarize(mean_count = mean(counts_lengthscaledtpm)) %>%
  ggplot(aes(x = genename,
             y = mean_count))
    geom_col(position = "dodge")
```
**Figure 3** 

```{r make-barplot-of-highly-expressed-genes, echo = FALSE}
# this code uses the same data as above, but use it to make a
# barplot - remember geom_col() is just like
# when you use geom_bar(stat = "identity)
joined_table %>%
  filter (genename == "") %>%
  summarize(mean_count = mean(counts_lengthscaledtpm)) %>%
  ggplot(aes(x = genename,
             y = mean_count,
             fill = normal_or_cancer)) +
    geom_col(position = "dodge")
```
**Figure 4** 

```{r make-barplot-of-highly-expressed-genes, echo = FALSE}
# this code uses the same data as above, but use it to make a
# barplot - remember geom_col() is just like
# when you use geom_bar(stat = "identity)
joined_table %>%
  filter (genename == "") %>%
  summarize(mean_count = mean(counts_lengthscaledtpm)) %>%
  ggplot(aes(x = genename,
             y = mean_count,
             fill = cancer_stage)) +
    geom_col(position = "dodge")
```
**Figure 5** 

```{r make-barplot-of-highly-expressed-genes, echo = FALSE}
# this code uses the same data as above, but use it to make a
# barplot - remember geom_col() is just like
# when you use geom_bar(stat = "identity)
joined_table %>%
  filter (genename == "") %>%
  summarize(mean_count = mean(counts_lengthscaledtpm)) %>%
  ggplot(aes(x = genename,
             y = mean_count,
             fill = smoking_status)) +
    geom_col(position = "dodge")
```
**Figure 6**
# Discussion

Add around 1-2 pages interpreting your results and considering future directions one might take in analyzing these data.

Look up what the genes encode for (what protein they made) and how this can lead to cancer

# Sources Cited
